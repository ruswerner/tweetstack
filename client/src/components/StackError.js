import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import SocialMoodBadIcon from 'material-ui/svg-icons/social/mood-bad';
import {red500} from 'material-ui/styles/colors';

export default class StackError extends PureComponent {

  static propTypes = {
    error: PropTypes.string.isRequired
  };

  render() {
    return (
      <div style={{
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
        flexGrow: 1,
        flexDirection: 'column',
        color: red500
      }}>
        <span style={{marginBottom: 8}}>{this.props.error}</span>
        <SocialMoodBadIcon style={{color: red500}}/>
      </div>
    );
  }

}
