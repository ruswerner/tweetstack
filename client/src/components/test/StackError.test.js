import React from 'react';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import StackError from '../StackError';
import renderer from 'react-test-renderer';

describe('StackError', function() {

  it('renders', () => {
    const tree = renderer.create(<MuiThemeProvider><StackError error={'random error'}/></MuiThemeProvider>).toJSON();
    expect(tree).toMatchSnapshot();
  });

});
