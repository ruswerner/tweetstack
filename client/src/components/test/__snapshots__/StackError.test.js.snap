// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StackError renders 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "color": "#f44336",
      "display": "flex",
      "flexDirection": "column",
      "flexGrow": 1,
      "justifyContent": "center",
    }
  }
>
  <span
    style={
      Object {
        "marginBottom": 8,
      }
    }
  >
    random error
  </span>
  <svg
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "color": "#f44336",
        "display": "inline-block",
        "fill": "currentColor",
        "height": 24,
        "muiPrepared": true,
        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "userSelect": "none",
        "width": 24,
      }
    }
    viewBox="0 0 24 24"
  >
    <path
      d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm3.5-9c.83 0 1.5-.67 1.5-1.5S16.33 8 15.5 8 14 8.67 14 9.5s.67 1.5 1.5 1.5zm-7 0c.83 0 1.5-.67 1.5-1.5S9.33 8 8.5 8 7 8.67 7 9.5 7.67 11 8.5 11zm3.5 3c-2.33 0-4.31 1.46-5.11 3.5h10.22c-.8-2.04-2.78-3.5-5.11-3.5z"
    />
  </svg>
</div>
`;
